examples/window-app/nrfconnect/main/WindowCovering.cpp

```diff
@@ -162,14 +143,29 @@ bool WindowCovering::TargetCompleted(MoveType aMoveType, NPercent100ths aCurrent
 
 void WindowCovering::StartTimer(MoveType aMoveType, uint32_t aTimeoutMs)
 {
-    if (aMoveType == MoveType::LIFT)
+    MoveType * moveType = chip::Platform::New<MoveType>();
+    VerifyOrReturn(moveType != nullptr);
+
+    *moveType = aMoveType;
+    (void) chip::DeviceLayer::SystemLayer().StartTimer(chip::System::Clock::Milliseconds32(aTimeoutMs), MoveTimerTimeoutCallback,
+                                                       reinterpret_cast<void *>(moveType));
+}
+
+void WindowCovering::MoveTimerTimeoutCallback(chip::System::Layer * systemLayer, void * appState)
+{
+    MoveType * moveType = reinterpret_cast<MoveType *>(appState);
+    VerifyOrReturn(moveType != nullptr);
+
+    if (*moveType == MoveType::LIFT)
     {
-        k_timer_start(&sLiftTimer, K_MSEC(sMoveTimeoutMs), K_NO_WAIT);
+        chip::DeviceLayer::PlatformMgr().ScheduleWork(WindowCovering::DriveCurrentLiftPosition);
     }
-    else if (aMoveType == MoveType::TILT)
+    else if (*moveType == MoveType::TILT)
     {
-        k_timer_start(&sTiltTimer, K_MSEC(sMoveTimeoutMs), K_NO_WAIT);
+        chip::DeviceLayer::PlatformMgr().ScheduleWork(WindowCovering::DriveCurrentTiltPosition);
     }
+
+    chip::Platform::Delete(moveType);
```

Comment:
I was just about to point out that we miss this deletion but the last rebase fixed it. That's why I am not a big fan of async calls parametrization in this fashion (hence the initial implementation of this fix from nrfconnect SDK with two separate callbacks) :)

Reply:
Hmm, I can see your point of view. But, this can be done in some ways. I agree that the way I did it was the most dangerous, and I forgot to free the memory. Still, I followed a suggestion that an argument can be passed within the pointer and in the same way it was done in the `SchedulePostAttributeChange` method, so I thought we should be coherent with that. Anyway, thanks for your comment here. :)